staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -49,6 +49,8 @@ type Interface interface {
 	LoadBalancer() (LoadBalancer, bool)
 	// Instances returns an instances interface. Also returns true if the interface is supported, false otherwise.
 	Instances() (Instances, bool)
+	// InstancesV2 is an external-only implementation for instances. Also returns true if the interface is supported, false otherwise.
```

Comment:
What do you mean by "external-only" ?

Reply:
Might it be internal-only? I'd like to ask for @andrewsykim 's original thought.

Reply:
Sorry, the comment in the example I shared was not fully thought out. 

"external-only" meant it should only be called by cloud-controller-manager at the moment. Cloud node controller in https://github.com/kubernetes/kubernetes/pull/91319 only runs in cloud-controller-manager and not in kube-controller-manager or kubelet.

Reply:
No sorry, your example helps me a lot to understand your thought. Updated the comments, PTAL